torch/quantization/fx/_equalize.py

```diff
@@ -698,3 +705,155 @@ def _convert_equalization_ref(model: GraphModule):
+
+    model_a_ns, model_b_ns = ns.add_loggers(
+        'fp32', model_a,
+        'int8', model_b,
+        ns.OutputLogger,
+        unmatchable_types_map=unmatchable_types_map
+    )
+
+    model_a_ns(x)
+    model_b_ns(x)
+
+    activation_comparison_dict = ns.extract_logger_info(
+        model_a_ns,
+        model_b_ns,
+        ns.OutputLogger,
+        'int8')
+    ns.extend_logger_results_with_comparison(
+        activation_comparison_dict,
+        'fp32', 'int8',
+        torch.quantization.ns.utils.compute_sqnr, 'sqnr'
+    )
+
+    # Construct a dictionary mapping layer names to the SQNR values
+    layer_sqnr_dict = {}
+    for key in activation_comparison_dict:
+        layer = activation_comparison_dict[key]['node_output']['int8'][0]['ref_node_name']
+        sqnr = activation_comparison_dict[key]['node_output']['int8'][0]['sqnr'][0]
+        layer_sqnr_dict[layer] = sqnr
+
+    return layer_sqnr_dict
+
+def _get_equalization_qconfig_dict(
+    layer_sqnr_dict: Dict[str, float],
+    model_b: nn.Module,
+    num_layers_to_equalize: int
+) -> Any:
+    """ Given the layer to SQNR dictionary, find the layers with the highest
+    quantization errors, and return an equalization_qconfig_dict
+    specifying to only equalize those top layers.
+
+    Args:
+        layer_sqnr_dict: Dictionary mapping layer names to SQNR values (found
+            when comparing an equalized model against a float model)
+        model_b: The equalized model used to construct the layer_sqnr_dict
+        num_layers_to_equalize: Number of layers with the highest quantization
+           errors to equalize
+    """
+
+    # Sort the layer_sqnr_dictionary values and get the layers with the lowest
+    # SQNR values (aka highest quantization errors)
+    layer_sqnr_sorted = sorted(layer_sqnr_dict.items(), key=lambda item: item[1])
+    layers_to_equalize = layer_sqnr_sorted[:num_layers_to_equalize]
+
+    # Trace the model so that we can find the accurate node_name_to_scope
+    from ..quantize_fx import QuantizationTracer
+    tracer = QuantizationTracer([], [])
+    tracer.trace(model_b)
+
+    # Constructs an equalization_qconfig_dict that specifies to only equalize
+    # the layers with the highest quantization errors
+    module_to_qconfig_list = list(
+        map(
+            lambda item: (
+                tracer.node_name_to_scope[item[0]][0],
+                default_equalization_qconfig,
+            ),
+            layers_to_equalize,
+        )
+    )
+
+    equalization_qconfig_dict = {"module_name": module_to_qconfig_list}
+    return equalization_qconfig_dict
+
+def selctive_equalization_with_numeric_suite(
+    float_model: nn.Module,
+    qconfig_dict: Any,
+    input: torch.Tensor,
+    num_layers_to_equalize: int,
+    prepare_custom_config_dict: Dict[str, Any] = None,
+    convert_custom_config_dict: Dict[str, Any] = None,
+) -> QuantizedGraphModule:
+    """ Selectively equalizes the given float model with the following steps:
```

Comment:
I think we should update the comment and code to say this is equalized + quantized output